observability/prometheus-backend/src/main/scala/sttp/client3/prometheus/PrometheusBackend.scala

```diff
@@ -60,6 +65,54 @@ object PrometheusBackend {
     )
   }
 
+  /** Add, if not present, label pairs for "method" and "status" (if Response is non-empty). In other words, if the user
+    * already supplied those labels, leave them as-is.
+    * @param config
+    *   user-supplied PrometheusBackend base collector
+    * @param req
+    *   Request
+    * @param maybeResp
+    *   Optional response
+    * @return
+    *   BaseCollectorConfig sub-type. Note that PrometheusBackend#apply and PrometheusListener's constructor reference
+    *   the sub-types of BaseCollectorConfig.
+    */
+  def addLabelPairs(config: BaseCollectorConfig, req: Request[_, _], maybeResp: Option[Response[_]]): config.T = {
```

Comment:
also, maybe a different signature, without the type member: `def addLabelPairs[T <: BaseCollectorConfig](cfg: T, ...)`?

Reply:
Yep - let me add that. `config.T` is a bit ugly, I agree.

Reply:
Done - https://github.com/softwaremill/sttp/pull/1651/commits/2e9728638835a51cb8fef88170bcc0f0b10d57ed. Not sure the `T#T` is any better. Let me know, please.